pkg/sql/sem/builtins/builtins.go

```diff
@@ -609,6 +610,41 @@ var builtins = map[string]builtinDefinition{
 		},
 	),
 
+	"gen_random_ulid": makeBuiltin(
+		tree.FunctionProperties{
+			Category: categoryIDGeneration,
+		},
+		tree.Overload{
+			Types:      tree.ArgTypes{},
+			ReturnType: tree.FixedReturnType(types.Uuid),
+			Fn: func(_ *tree.EvalContext, _ tree.Datums) (tree.Datum, error) {
+				t := time.Now().UnixNano()
+				entropy := ulid.Monotonic(rand.New(rand.NewSource(t)), 0)
+				uv := ulid.MustNew(ulid.Timestamp(time.Unix(0, t)), entropy)
+				return tree.NewDUuid(tree.DUuid{UUID: uuid.UUID(uv)}), nil
+			},
+			Info:       "Generates a random ULID and returns it as a value of UUID type.",
+			Volatility: tree.VolatilityVolatile,
+		},
+	),
+
+	"uuid_ulid_to_string": makeBuiltin(defProps(),
```

Comment:
I was thinking something like `uuid_to_ulid` or `uid_to_ulid_string` makes more sense to me.

Reply:
The intention was to make clear that you were casting an "uuid-formmated" ulid, but I get the point. If you feel that we won't need this name in the future I would go for the short form `uuid_to_ulid`.

Reply:
Yeah I understand what you were going for, I think as long as were clear in the description of the function, it's fine. Technically any UUID can be turned into a ULID but users should know what the function is used for by the description.

I think `uuid_to_ulid` is fine.

Reply:
Cool! I'll got for it then.